changeFeedSamples/trackingChangesToBlobs/src/main/java/ChangeFeedTimer.java

```diff
@@ -0,0 +1,169 @@
+
+        // Create a Timer
+        Timer timer = new Timer();
+        TimerTask task = new ChangeFeedHelper(blobContainerClient, blobClient, changefeedClient, cursor);
+
+        // Running on schedule
+        timer.scheduleAtFixedRate(task, 0, interval);
+    }
+
+    /**
+     * Retrieves cursor from container in case where cursor exists. If not, returns null
+     */
+    public static String getCursor(BlobClient blobClient) {
+        // Checking if blob exists to see if there is an existing cursor
+        if (blobClient.exists()) {
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            blobClient.download(outputStream);
+            return new String(outputStream.toByteArray());
+        }
+        return null;
+    }
+}
+
+class ChangeFeedHelper extends TimerTask {
+    public String cursor;
+    public BlobContainerClient containerClient;
+    public BlobClient blobClient;
+    public BlobChangefeedClient changefeedClient;
+
+    // Filtering
+    String trackedContainer = "containers/test-changefeed-container";
+    String trackedBlob = "exampleBlob.txt";
+    String eventType = "BlobCreated";
+    Predicate<BlobChangefeedEvent> checkContainerName = (event) -> event.getSubject().contains(trackedContainer);
+    Predicate<BlobChangefeedEvent> checkBlobName = (event) -> event.getSubject().contains(trackedBlob);
+    Predicate<BlobChangefeedEvent> checkEventType = (event) -> event.getEventType().toString().equals(eventType);
+
+    public ChangeFeedHelper(BlobContainerClient containerClient, BlobClient blobClient, BlobChangefeedClient changefeedClient) {
+        this(containerClient, blobClient, changefeedClient, null);
+    }
+
+    public ChangeFeedHelper(BlobContainerClient containerClient, BlobClient blobClient, BlobChangefeedClient changefeedClient, String cursor) {
+        this.containerClient = containerClient;
+        this.blobClient = blobClient;
+        this.changefeedClient = changefeedClient;
+        this.cursor = cursor;
+    }
+
+    /**
+     * Every time the timer executes, check for new events starting from where cursor points at and filter events. Filter
+     * depends on how the predicates are set up.
+     */
+    public void run() {
+        BlobChangefeedPagedIterable iterable = null;
+
+        // Starting from cursor position if there exists one
+        if(this.cursor == null) {
+            iterable = this.changefeedClient.getEvents();
+        }
+        else {
+            iterable = this.changefeedClient.getEvents(this.cursor);
+        }
+        Iterable<BlobChangefeedPagedResponse> pages = iterable.iterableByPage(1);
+
+        System.out.println("Printing all events satisfying filters");
+        // Returning event info if event matches blob name
+        for (BlobChangefeedPagedResponse page : pages) {
+            List<BlobChangefeedEvent> iter = page.getValue();
+            for (BlobChangefeedEvent event : iter) {
```

Comment:
Instead of using a for each loop like this, consider using streams

https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html